File: app/src/main/java/fr/geobert/radis/service/OnRefreshReceiver.java
package fr.geobert.radis.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import fr.geobert.radis.tools.UpdateDisplayInterface;

public class OnRefreshReceiver extends BroadcastReceiver {

    private UpdateDisplayInterface mActivity;

    public OnRefreshReceiver(UpdateDisplayInterface activity) {
        super();
        mActivity = activity;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        mActivity.updateDisplay(intent);
    }

}

File: app/src/main/kotlin/fr/geobert/radis/service/OnRefreshReceiver.kt
package fr.geobert.radis.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import fr.geobert.radis.tools.UpdateDisplayInterface

class OnRefreshReceiver(private val mActivity: UpdateDisplayInterface) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        mActivity.updateDisplay(intent)
    }

}
